backend/FwLite/FwLiteDesktop/Platforms/Windows/AppUpdateService.cs

```diff
@@ -0,0 +1,117 @@
+
+        await ApplyUpdate(response.Release);
+    }
+
+    private async Task ApplyUpdate(FwLiteRelease latestRelease)
+    {
+        logger.LogInformation("New version available: {Version}", latestRelease.Version);
+        var packageManager = new PackageManager();
+        var asyncOperation = packageManager.AddPackageAsync(new Uri(latestRelease.Url), [], DeploymentOptions.None);
+        asyncOperation.Progress = (info, progressInfo) =>
+        {
+            logger.LogInformation("Downloading update: {ProgressPercentage}%", progressInfo.percentage);
+        };
+        var result = await asyncOperation.AsTask();
+        if (!string.IsNullOrEmpty(result.ErrorText))
+        {
+            logger.LogError(result.ExtendedErrorCode, "Failed to download update: {ErrorText}", result.ErrorText);
+            return;
+        }
+
+        logger.LogInformation("Update downloaded, will install on next restart");
+    }
+
+    private async Task<ShouldUpdateResponse> ShouldUpdate()
+    {
+        try
+        {
+            var response = await httpClientFactory
+                .CreateClient("Lexbox")
+                .SendAsync(new HttpRequestMessage(HttpMethod.Get, ShouldUpdateUrl)
+                {
+                    Headers = { { "User-Agent", $"Fieldworks-Lite-Client/{AppVersion.Version}" } }
+                });
+            if (!response.IsSuccessStatusCode)
+            {
+                var responseContent = await response.Content.ReadAsStringAsync();
+                logger.LogError("Failed to get should update response lexbox: {StatusCode} {ResponseContent}",
+                    response.StatusCode,
+                    responseContent);
+                return new ShouldUpdateResponse(null);
+            }
+
+            return await response.Content.ReadFromJsonAsync<ShouldUpdateResponse>() ?? new ShouldUpdateResponse(null);
+        }
+        catch (Exception e)
+        {
+            if (connectivity.NetworkAccess == NetworkAccess.Internet)
+            {
+                logger.LogError(e, "Failed to fetch latest release");
+            }
+            else
+            {
+                logger.LogInformation(e, "Failed to fetch latest release, no internet connection");
+            }
+
+            return new ShouldUpdateResponse(null);
+        }
+    }
+
+    private bool ShouldCheckForUpdate()
+    {
+        if (ValidPositiveEnvVarValues.Contains(Environment.GetEnvironmentVariable(ForceUpdateCheckEnvVar) ?? ""))
+            return true;
+        var lastChecked = preferences.Get(LastUpdateCheck, DateTime.MinValue);
+        var timeSinceLastCheck = DateTime.UtcNow - lastChecked;
+        //if last checked is in the future (should never happen), then we want to reset the time and check again
+        if (timeSinceLastCheck.Hours < -1)
+        {
+            preferences.Set(LastUpdateCheck, DateTime.UtcNow);
+            return true;
+        }
+        if (timeSinceLastCheck.Hours < 20) return false;
```

Comment:
You're right is is cheap to check. At this point I have no idea what a failed update would look like or how that would even happen considering we're not running any custom code during install. Not that I doubt it could happen. That said, tracking update attempts introduces complexity which I'd rather avoid, so maybe let's leave it as is for now and see what we learn with people actually using it and updating to new versions